Tests/SourceKitLSPTests/ExecuteCommandTests.swift

```diff
@@ -199,110 +199,127 @@ final class ExecuteCommandTests: XCTestCase {
+          files: files,
+          manifest: SwiftPMTestProject.macroPackageManifest,
+          capabilities: ClientCapabilities(experimental: [
+            "workspace/peekDocuments": .bool(peekDocuments),
+            "workspace/getReferenceDocument": .bool(getReferenceDocument),
+          ]),
+          options: options,
+          enableBackgroundIndexing: true
         )
 
-        let metadata = SourceKitLSPCommandMetadata(textDocument: TextDocumentIdentifier(uri))
+        let (uri, positions) = try project.openDocument("MyMacroClient.swift")
 
-        var command = args.asCommand()
-        command.arguments?.append(metadata.encodeToLSPAny())
+        let positionMarkersToBeTested = [
+          (start: "1️⃣", end: "1️⃣"),
+          (start: "2️⃣", end: "2️⃣"),
+          (start: "1️⃣", end: "3️⃣"),
+          (start: "2️⃣", end: "3️⃣"),
+        ]
+
+        for positionMarker in positionMarkersToBeTested {
+          let args = ExpandMacroCommand(
+            positionRange: positions[positionMarker.start]..<positions[positionMarker.end],
+            textDocument: TextDocumentIdentifier(uri)
+          )
 
-        let request = ExecuteCommandRequest(command: command.command, arguments: command.arguments)
+          let metadata = SourceKitLSPCommandMetadata(textDocument: TextDocumentIdentifier(uri))
 
-        if peekDocuments {
-          let expectation = self.expectation(description: "Handle Peek Documents Request")
-          let peekDocumentsRequestURIs = ThreadSafeBox<[DocumentURI]?>(initialValue: nil)
+          var command = args.asCommand()
+          command.arguments?.append(metadata.encodeToLSPAny())
 
-          project.testClient.handleSingleRequest { (req: PeekDocumentsRequest) in
-            peekDocumentsRequestURIs.value = req.locations
-            expectation.fulfill()
-            return PeekDocumentsResponse(success: true)
-          }
+          let request = ExecuteCommandRequest(command: command.command, arguments: command.arguments)
 
-          _ = try await project.testClient.send(request)
+          if peekDocuments && getReferenceDocument {
+            let expectation = self.expectation(description: "Handle Peek Documents Request")
+            let peekDocumentsRequestURIs = ThreadSafeBox<[DocumentURI]?>(initialValue: nil)
 
-          try await fulfillmentOfOrThrow([expectation])
+            project.testClient.handleSingleRequest { (req: PeekDocumentsRequest) in
+              peekDocumentsRequestURIs.value = req.locations
+              expectation.fulfill()
+              return PeekDocumentsResponse(success: true)
+            }
 
-          let urls = try XCTUnwrap(
-            peekDocumentsRequestURIs.value?.map {
-              return try XCTUnwrap(
-                $0.fileURL,
+            _ = try await project.testClient.send(request)
+
+            try await fulfillmentOfOrThrow([expectation])
+
+            let urlStrings = try XCTUnwrap(
```

Comment:
I think we could save a bit of casting if this was just `[DocumentURI]`.

Same in the test below.